app/src/main/java/co/nimblehq/ui/base/BaseViewModel.kt

```diff
@@ -32,5 +36,46 @@ abstract class BaseViewModel : ViewModel() {
     val navigator: Observable<NavigationEvent>
         get() = _navigator
 
+    /**
+     * To show loading manually, should call `hideLoading` after
+     */
+    protected fun showLoading() {
+        if (loadingCount == 0) {
+            _showLoading.onNext(true)
+        }
+        loadingCount++
+    }
+
+    /**
+     * To hide loading manually, should be called after `showLoading`
+     */
+    protected fun hideLoading() {
+        loadingCount--
+        if (loadingCount == 0) {
+            _showLoading.onNext(false)
+        }
+    }
```

Comment:
@minhnimble possible, but the main purpose of this PR is trying to build a common mechanism to show/hide loading progress, I'm wondering about the `forceHideLoading` method will break the rule, instead of trying to figure out whether we missed any show/hide actions, we try to use that method to speed up the dev progress.